dlt/pipeline/pipeline.py

```diff
@@ -877,6 +882,7 @@ def _extract_source(self, storage: ExtractorStorage, source: DltSource, max_para
             pipeline_schema.update_schema(
                 pipeline_schema.normalize_table_identifiers(table)
             )
+            pipeline_schema.set_schema_contract_settings(source_schema._settings.get("schema_contract_settings", {}))
```

Comment:
why do you update this in a loop? once is enough. also it seems we need a method to merge two schemas - that would merge schema settings and properties uh.

Reply:
I added this method